com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/clients/RecentLogsRetriever.java

```diff
@@ -1,164 +1,75 @@
 package com.sap.cloud.lm.sl.cf.core.cf.clients;
 
-import java.io.ByteArrayOutputStream;
-import java.io.IOException;
-import java.io.InputStream;
-import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Date;
+import java.util.Collections;
 import java.util.List;
-import java.util.UUID;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
-import javax.inject.Inject;
 import javax.inject.Named;
 
-import org.apache.commons.fileupload.MultipartStream;
+import com.sap.cloud.lm.sl.cf.core.Constants;
+import com.sap.cloud.lm.sl.cf.core.util.ImmutableLogsOffset;
+import com.sap.cloud.lm.sl.cf.core.util.LogsOffset;
 import org.cloudfoundry.client.lib.CloudControllerClient;
 import org.cloudfoundry.client.lib.domain.ApplicationLog;
-import org.springframework.core.io.Resource;
-import org.springframework.http.HttpMethod;
-import org.springframework.http.MediaType;
-import org.springframework.http.ResponseEntity;
-
-import com.google.protobuf.InvalidProtocolBufferException;
-import com.sap.cloud.lm.sl.cf.client.events.EventFactory;
-import com.sap.cloud.lm.sl.cf.client.events.LogFactory.LogMessage;
-import com.sap.cloud.lm.sl.cf.client.events.LogFactory.LogMessage.MessageType;
-import com.sap.cloud.lm.sl.cf.core.message.Messages;
-import com.sap.cloud.lm.sl.common.SLException;
+import org.flowable.variable.api.delegate.VariableScope;
 
 @Named
-public class RecentLogsRetriever extends CustomControllerClient {
-
-    public static final String RECENT_LOGS_ENDPOINT = "/apps/{guid}/recentlogs";
+public class RecentLogsRetriever {
 
     private Supplier<CustomControllerClientErrorHandler> errorHandlerFactory = CustomControllerClientErrorHandler::new;
 
-    @Inject
-    public RecentLogsRetriever(RestTemplateFactory restTemplateFactory) {
-        super(restTemplateFactory);
-    }
-
     RecentLogsRetriever withErrorHandlerFactory(Supplier<CustomControllerClientErrorHandler> errorHandlerFactory) {
         this.errorHandlerFactory = errorHandlerFactory;
         return this;
     }
 
-    public List<ApplicationLog> getRecentLogsSafely(CloudControllerClient client, String appName) {
+    public List<ApplicationLog> getRecentLogsSafely(CloudControllerClient client, String appName, VariableScope context) {

```

Comment:
This class should not know about Flowable. Don't use `VariableScope` here.

Reply:
Where should the filtering be then?